engine/src/test/java/io/camunda/zeebe/engine/state/migration/to_1_3/TemporaryVariableMigrationTest.java

```diff
@@ -128,5 +139,30 @@ public void afterMigrationRunNoFurtherMigrationIsNeeded() {
       assertThat(eventTrigger.getElementId())
           .isEqualTo(BufferUtil.wrapString(String.format("migrated-variable-%d", EVENT_SCOPE_KEY)));
     }
+
+    @Test
+    public void eventSubProcessGetsMigratedCorrectly() {
+      // given
+      final long flowScopeKey = 200L;
+      final ProcessInstanceRecord processInstanceRecord = new ProcessInstanceRecord();
+      processInstanceRecord.setBpmnElementType(BpmnElementType.EVENT_SUB_PROCESS);
+      processInstanceRecord.setFlowScopeKey(flowScopeKey);
+      elementInstanceState.newInstance(
+          EVENT_SCOPE_KEY, processInstanceRecord, ProcessInstanceIntent.ELEMENT_ACTIVATED);
+
+      // when
+      sutMigration.runMigration(zeebeState);
+
+      // then
+      final EventTrigger oldEventTrigger =
+          zeebeState.getEventScopeInstanceState().peekEventTrigger(EVENT_SCOPE_KEY);
+      final EventTrigger newEventTrigger =
+          zeebeState.getEventScopeInstanceState().peekEventTrigger(flowScopeKey);
+      assertThat(oldEventTrigger).isNull();
+      assertThat(newEventTrigger).isNotNull();
+      assertThat(newEventTrigger.getEventKey()).isEqualTo(-1L);
+      assertThat(newEventTrigger.getElementId())
+          .isEqualTo(BufferUtil.wrapString(String.format("migrated-variable-%d", EVENT_SCOPE_KEY)));
+    }
```

Comment:
:wrench: We should verify that the temporary variables are removed, like in the other test case. Just to be sure.